internal/provider/provider_test.go

```diff
@@ -14,12 +12,3 @@ import (
 var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServer, error){
 	"mezmo": providerserver.NewProtocol6WithError(New("test")()),
 }
-
-// Terraform uses a flag to prevent tests from running unintentionally,
-// as Terraform resources are often linked to real-world resources and infrastructure.
-// In our case, we support running the integration tests against a pipeline-service container.
-// We could support running against dev / staging in the future, once we expose the pipeline-service
-// behind the Gateway.
-func init() {
-	os.Setenv("TF_ACC", "1")
```

Comment:
I understand the desire to change the test names to `Acc`, but I'm not sure I get what we're gaining by only setting this in the compose file.  We *always* need the backend service to run these tests, even locally.  Do we even have a use case for unit tests?  Aside from error cases, what can be unit tested?  And what's the harm of those tests running with `TF_ACC=1` and failing early?

I'm just trying to understand the value.  If I'm just setting this myself on every run (because I always run integration tests), it's just feels like more work when developing locally.

Reply:
We have a few unit tests and I made some which I used to understand the code a bit better. I did this mainly for the speedup, we can run the unit tests in less than a second versus waiting almost 3 minutes for the acceptance tests. Makes for a faster feedback loop. 

```
go test -v ./...  0.74s user 0.92s system 359% cpu 0.460 total
TF_ACC=1 go test -v ./...  171.43s user 66.67s system 318% cpu 1:14.73 total
```__

Reply:
I'm not arguing the speed, I'm just saying that 99% of our tests are integration tests that are regularly run during local development.  Spinning up compose to run those is much slower when testing repeatedly, so now we'll have to add the env var on the command line.

We can keep it, it's just a dev flow adjustment.

Reply:
Perhaps a good compromise is adding `TF_ACC=1` to `local.env` and running the integration tests with `npm run local -- go test -v -run whatever`

We can do that later.